pkg/cloud/vsphere/provisioner/common/templates.go

```diff
@@ -489,10 +502,20 @@ KUBEADM=$(getversion kubeadm ${KUBELET_VERSION}-)
 
 apt-get install -y docker.io=${DOCKER_VER}
 
-# Explicit cni version is a temporary workaround till the right version can be automatically detected correctly
-apt-get install -y \
-    kubelet=${KUBELET} \
-    kubeadm=${KUBEADM}
+### TEMPORARY solution
+# https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/238
+# Currently, ubuntu packaging includes kubernetes-cni 0.7.5 when performing
+# apt-get install kubernetes-cni.  Older versions of k8s complains and this
+# script fails.  This sill install 0.7.5 for k8s 1.14 and higher.  It will
+# fall back to 0.6.0 for older versions.
+
+if [[ "${KUBELET_VERSION}" -ge "1.14" ]]; then
+	apt-get install -y kubernetes-cni=0.7.5-00*
+else
+	apt-get install -y kubernetes-cni=0.6.0-00*
```

Comment:
lets do `apt-get install -y kubernetes-cni=0.6.*` instead so that future backports can be included